tests/manage/z_cluster/cluster_expansion/conftest.py

```diff
@@ -107,3 +110,40 @@ def factory(ocs_nodes=False, node_count=3, taint_label=None):
         log.info(f"Successfully Tainted nodes {new_nodes} with {taint_label}")
 
     return factory
+
+
+@pytest.fixture
+def add_capacity_setup(add_nodes):
+    """
+    Check that we have the right configurations before we start the test
+    """
+    log.info("Start add capacity setup")
+    osd_pods_before = get_osd_pods()
+    number_of_osd_pods_before = len(osd_pods_before)
+    if number_of_osd_pods_before >= constants.MAX_OSDS:
+        pytest.skip("We have maximum of OSDs in the cluster")
+
+    # If we use vSphere we may need to add more worker nodes
+    # to the cluster before starting the test
+    if (
+        config.ENV_DATA["platform"].lower() == constants.VSPHERE_PLATFORM
+        and number_of_osd_pods_before >= 9
+    ):
+        num_of_expected_wnodes = 6
+        wnodes = get_worker_nodes()
+        num_of_wnodes = len(wnodes)
+        logging.info(
+            f"We have {number_of_osd_pods_before} OSDs in the cluster, "
+            f"and {num_of_wnodes} worker nodes in the cluster"
+        )
+        if num_of_wnodes < num_of_expected_wnodes:
+            num_of_wnodes_to_add = num_of_expected_wnodes - num_of_wnodes
+            logging.info(
+                f"Adding more {num_of_wnodes_to_add} worker nodes to the cluster"
+            )
+            add_nodes(ocs_nodes=True, node_count=num_of_wnodes_to_add)
+
+        wnodes_not_in_ocs = get_worker_nodes_not_in_ocs()
+        if wnodes_not_in_ocs:
+            logging.info("Label the worker nodes that are not in OCS")
+            label_nodes(wnodes_not_in_ocs)
```

Comment:
Why we are adding label for non ocs node this step should be optional

Reply:
All the 6 worker nodes should be OCS labeled, so we can use them for the newly created OSDs.

Reply:
dont 144 line do for us create new node and label it for OCS

Reply:
Line 144 adds new nodes and OCS labels them. But if we already have worker nodes(before starting the test) that are not labeled - then we need this line to label them.

Reply:
For example, let's say we have 3 OCS worker nodes, and 2 worker nodes not labeled. The fixture will add one worker node and to reach 6 worker nodes and OCS labeled the node. Then it will take the other 2 worker nodes that are not labeled, and OCS labeled them.